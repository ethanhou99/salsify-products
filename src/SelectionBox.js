import React, { Component } from 'react';
import PropTypes from 'prop-types'

class SelectionBox extends Component {
  static propTypes = {
    items: PropTypes.array.isRequired,
    defaultValue: PropTypes.string.isRequired,
    func: PropTypes.func.isRequired
  }
  
  render() {
    const {items, defaultValue, func} = this.props;
    var itemList = items.map(func)
    return (
      <div>
        <select>
          <option value=''>{defaultValue}</option>
          { itemList }
        </select>
      </div>
    )
  }
}

export default SelectionBox;