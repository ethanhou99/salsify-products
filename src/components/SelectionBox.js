import React, { Component } from 'react';
import PropTypes from 'prop-types'

class SelectionBox extends Component {
  static propTypes = {
    items: PropTypes.array.isRequired,
    defaultVal: PropTypes.string.isRequired,
    func: PropTypes.func.isRequired
  }

  constructor(props) {
    super(props);
    this.state = {
      filterValue: ''
    }
  }
  
  handleChange = (e) => {
    if (e.target.value !== '') {
      if (this.props.funcType === 'property') {
          this.props.setType(this.getDataType(e.target.value).type);
          this.props.onUpdate(e.target.value);
          this.setState({filterValue: e.target.value});
      } else if (this.props.funcType === 'operator') {
        this.props.onUpdate(e.target.value);
        this.setState({filterValue: e.target.value});
      } else {
        this.props.onUpdate('');
        this.setState({filterValue: ''});
      }
    }
  };

  getDataType = (e) => {
    return this.props.items.filter(property => property.name === e)[0]
  }

  render() {
    const {items, defaultVal, func} = this.props;
    var itemList = items.map(func)
    return (
      <select className='select-box-style' 
        onChange={this.handleChange} 
        value={this.state.filterValue}>
          <option>{defaultVal}</option>
          { itemList }
      </select>
    )
  }
}

export default SelectionBox;