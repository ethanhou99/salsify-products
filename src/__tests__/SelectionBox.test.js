import React from 'react';
import {cleanup, fireEvent, render} from '@testing-library/react';
import SelectionBox from '../components/SelectionBox';

// automatically unmount and cleanup DOM after the test is finished.
afterEach(cleanup);

const defaultProps = { 
  items: ['Product Name', 'color', 'weight (oz)', 'category', 'wireless'],
  defaultVal: 'Select a Property',
  func: jest.fn()
};

describe('Selection Box test', () => {
  it('Selection box should render correctly', () => {
    const component = render(<SelectionBox {...defaultProps}/>)
    expect(component.firstChild).toMatchSnapshot()
   })

   it('User should able to select from the selection box', () => {
    const {getByText} = render(<SelectionBox {...defaultProps}/>);
    fireEvent.select(getByText('Select a Property'));
    expect(getByText('Select a Property')).toBeTruthy();
  });
});