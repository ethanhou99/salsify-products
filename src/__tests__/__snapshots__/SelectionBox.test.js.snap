// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Selection Box test Selection box should render correctly 1`] = `undefined`;
